src/components/structures/RoomView.js

```diff
@@ -534,30 +560,19 @@ module.exports = React.createClass({
 
     // set off a pagination request.
     onMessageListFillRequest: function(backwards) {
-        if (!backwards)
+        var dir = backwards ? EventTimeline.BACKWARDS : EventTimeline.FORWARDS;
+        if(!this._timelineWindow.canPaginate(dir)) {
```

Comment:
Do we need to null-guard `this._timelineWindow`? It is initially `null` and is only set when the `Room` object resolves. `ScrollView` won't know when this is presumably so will be firing off `onMessageListFillRequests` willy-nilly.